statchatui/src/Stats/Stats.jsx

```diff
@@ -1,25 +1,44 @@
 import React from "react";
 
-const Stats = ({ statData }) => {
+const Stats = ({ statData, gameNames, formData }) => {
   if (!statData) {
-    return <div>Stats not yet fetched.</div>;
+    return <div>{"Stats not yet fetched."}</div>;
   }
-
   let kills, level, name, rank;
-  kills = statData.total.kills.value;
-  level = statData.global.level;
-  name = statData.global.name;
-  rank = statData.global.rank.rankName;
+  if (statData.status === 404) {
+    return <div>Error: {statData.errors[0].message}</div>;
+    // this handles the error if the r6 username does not exist
+  }
+  if (statData.Error) {
+    return <div>Error: {statData.Error}</div>;
+  }
+  if (formData.gameName === gameNames.APEX) {
+    if (statData.total) {
+      kills = statData.total.kills.value;
+    }
+    if (statData.global) {
+      level = statData.global.level;
+      name = statData.global.name;
+      rank = statData.global.rank.rankName;
+    }
+  } else if (formData.gameName === gameNames.R6) {
+    if (statData.data) {
+      name = statData.data.metadata.user;
+      level = statData.data.metadata.level;
+      rank = statData.data.stats_general.rank.mmr;
+      kills = statData.data.stats_general.kills;
```

Comment:
This looks good! Just an idea to make your code simpler:

1. Instead of calling your game APIs directly here, you can call them from your server.
2. Your server can look at and rearrange the data so everything is in the same format, no matter what game it is. Something like this:

```
  {
    game: 'apex',
    data: {
      kills: 1000,
      rank: '....',
      level: '...',
    } 
  }
```

Then in your UI you can delete basically all of this code :D 
Lower priority, but something you can do to help make your code nicer and more maintainable.

Reply:
i will add this to a low priority fixme